assets/src/components/detours/detourRouteSelectionPanel.tsx

```diff
@@ -86,7 +88,7 @@ export const DetourRouteSelectionPanel = ({
 
           <section className="pb-3">
             <h2 className="c-diversion-panel__h2">Choose direction</h2>
-            {selectedRouteInfo.selectedRoute ? (
+            {selectedRouteInfo.routePatterns ? (
```

Comment:
nit(non-blocking): if you wrapped the RPC in this check, and left the `selectedRoute` check, then we'd properly display nothing when a route _is_ selected and then only show the RPC after it's loaded. This then makes the change to add a loading spinner while `routePatterns` is loading easier, but it's small enough that that this is fine to leave as is for now.

Reply:
Is [this](https://github.com/mbta/skate/pull/2647/commits/f5b33ec89b27e206ab880710157fb1cd37dd09d7) what you had in mind?

Reply:
lgtm